packages/api-v2/src/ws-client/request.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    TSocketResponseData,
+    TSocketResponse,
+    TSocketEndpointNames,
+    TSocketRequestPayload,
+    TSocketSubscribableEndpointNames,
+} from '../../types';
+
+const REQ_TIMEOUT = 20000;
+
+// sequence number for requests
+let reqSeqNumber = 0;
```

Comment:
It worries me that it's essentially a global, although even if request was called with two different websockets, I think it's fine that the other websocket skips a few increments. But shouldn't this be a factory, `createRequester` then you can keep the `reqSeqNumber` with the instance?

Reply:
yeah I haven't been super happy with that pattern, 
maybe move it to ws-client as this the the class responsible for holding majority of state?

Reply:
in particula, ws-client holds connnection,
and req_id is generally per-connection state,